riptide-failsafe/src/main/java/org/zalando/riptide/failsafe/FailsafePlugin.java

```diff
@@ -20,113 +20,105 @@
 import org.zalando.riptide.RequestArguments;
 import org.zalando.riptide.RequestExecution;
 
-import javax.annotation.Nullable;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.ScheduledExecutorService;
+import java.util.function.Predicate;
+import java.util.function.UnaryOperator;
+import java.util.stream.Stream;
 
-import static java.util.Objects.requireNonNull;
 import static lombok.AccessLevel.PRIVATE;
 import static org.apiguardian.api.API.Status.MAINTAINED;
 import static org.zalando.riptide.CancelableCompletableFuture.forwardTo;
 import static org.zalando.riptide.CancelableCompletableFuture.preserveCancelability;
 
 @API(status = MAINTAINED)
-@AllArgsConstructor
-public final class FailsafePlugin {
+@AllArgsConstructor(access = PRIVATE)
+public final class FailsafePlugin implements Plugin {
 
+    private final ImmutableList<? extends Policy<ClientHttpResponse>> policies;
     private final ScheduledExecutorService scheduler;
+    private final MethodDetector idempotent;
+    private final RetryListener listener;
+
+    public FailsafePlugin(final ImmutableList<? extends Policy<ClientHttpResponse>> policies,
+            final ScheduledExecutorService scheduler) {
+        this(policies, scheduler, MethodDetector.compound(
```

Comment:
A bit weird that one can construct a `FailsafePlugin` with a specific `MethodDector` and/or `RetryListener` without any `Policy`. I guess allowing such "null" `FailsafePlugin` allows for easier configuration in the AutoConfig?